src/aggregator/aggregator/options.go

```diff
@@ -387,6 +387,11 @@ type options struct {
 
 // NewOptions create a new set of options.
 func NewOptions() Options {
+	return NewOptionsWithClock(clock.NewOptions())
```

Comment:
Do we still need this parameterless `NewOptions()` (I thought the goal was to avoid it)?
If it is used for the tests, maybe we could have `NewTestOptions()` for test code, and then `NewOptions(clockOpts clock.Options)` would always require the argument?

Reply:
I kept it for consistency, since in all the other cases `NewOptions()` function has no arguments. Though I don't really have strong opinion on this.

Reply:
Don't think there is a reason why we should avoid it having arguments. This is dependency injection, and sometimes dependencies can have other dependencies (`Options` is a poor name for this, IMHO).
So are there any use cases for the parameterless `NewOptions()`?

Reply:
Most almost all usages were in tests. Moved the parameter to `NewOptions()` https://github.com/m3db/m3/commit/82dabe6f50a028e9d2d804db404d27e15c35d8bf#diff-53adbc2f22f64078cc058df41003a26a6a902996fcb19bc2321d6bc06bdec7bfL389-R389